ansible/roles/test/tasks/port_alias/interface_config.yml

```diff
@@ -0,0 +1,75 @@
+#set the test interface according to default or alias mode 
+- set_fact: 
+    intf: "{{interface_alias if (mode=='alias') else interface}}"
+
+- set_fact:
+    native_speed: "{{port_speed[interface_alias]}}"
+
+- block: 
+
+    - name: shutdown the interface {{intf}} in {{mode}} mode 
+      shell: sudo config interface shutdown {{intf}}
+      register: out
+      failed_when: out.rc != 0
+
+    - pause: seconds=3
+
+    - name:  Get interface status
+      show_interface: command="status"  interfaces={{intf}}
+
+    - pause: seconds=3
+
+    - name: Check  whether the status  is down
+      assert: {that: "'{{int_status[intf]['admin_state']}}' == 'down'"}
+
+    - name: Bringup the interface  {{intf}} in {{mode}} mode
+      shell: sudo config interface startup  {{intf}}
+      register: out
+      failed_when: out.rc != 0
+    
+    - pause: seconds=3
+    - name:  Get interface status
+      show_interface: command="status"  interfaces="{{intf}}"
+      
+    - name: Check  whether the status  is up
+      assert: {that: "'{{int_status[intf]['admin_state']}}' == 'up'"}
+
+# check the  config interface speed
+
+    - name: configure interface speed to 10G in {{mode}} mode
+      shell: sudo config interface speed {{intf}} 10000
+      register: out
+      failed_when: out.rc != 0
+
+    - name: get the interface speed
+      shell: sudo redis-cli -n 4 HGET "PORT|{{interface}}" 'speed'
+      register: speed
+
+    - debug: var=speed
+
+    - name: Check  whether the speed is set to 10G
+      assert: {that: "'{{speed.stdout}}' == '10000'"}
+
+    - name: chamge interface speed to native speed and check
+      shell: sudo config interface speed {{intf}} {{native_speed}}
+      register: out
+      failed_when: out.rc != 0
+
+    - name: get the interface speed
+      shell: sudo  redis-cli -n 4 HGET "PORT|{{interface}}" 'speed'
+      register: speed
+
+    - name: Check  whether the speed is set to native speed
+      assert: {that: "'{{speed.stdout}}' == '{{native_speed}}'"}
+
+  environment:
+      SONIC_CLI_IFACE_MODE: "{{ifmode}}"
+
+  always:
+
+    - name: set the interface up 
+      shell:  sudo config interface startup  {{intf}} 
```

Comment:
Please note that the "config interface ..." syntax changed recently. The interface name should now be placed after "interface" and before the subcommand. Please update all "config interface ..." commands and confirm they work properly.

Reply:
@chitra-raghavan: Can you or @paavaanan please investigate the root cause of https://github.com/Azure/SONiC/issues/268?